opkg: patch: don't filter out directories when claiming ownership of file

BKPepe opened the pull request:
Directories are now owned by a particular package, which ensures their
deletion in the event that a package leaves an empty directory during package
removal.

Fixes: https://github.com/openwrt/openwrt/issues/7519, https://github.com/openwrt/packages/issues/10136, https://gitlab.nic.cz/turris/os/packages/-/issues/576

Sent via emailing list for [opkg repository](https://git.openwrt.org/?p=project/opkg-lede.git;a=summary):
- https://lists.openwrt.org/pipermail/openwrt-devel/2022-May/038613.html

paper42 committed:
opkg: don't filter out directories when claiming ownership of a file

Directories are now owned by a particular package, which ensures their
deletion in the event that a package leaves an empty directory during package
removal.

Signed-off-by: Michal Vasilek <michal.vasilek@nic.cz>

Ansuel commented:
@BKPepe I edited the message to link the issue... It seems github doesn't like the list way and the only correct way is use a long string with ',' as a separator...

BKPepe commented:
Thanks! :) I think I might be mistaken, but still, these issues need to be closed manually as pull requests will be marked as closed because OpenWrt does not use fast-forward merge, and the commit itself will reference those issues instead of closing them.

Ansuel commented:
For issue in the repo they gets autoclosed.

This repo is a mirror of git, when it's refreshed, the issue is autoclosed as the commit will have that tag in the description.

Ansuel commented:
@BKPepe it's a pitty that we lose track of it tho (as in git we will have just the commit not the pr) but i understand the concern...

BKPepe commented:
Not at this time, as in the patch, @paper42 mentions why it is necessary and which issue it fixes, but it would be nice to have this merged and included in OpenWrt 22.03, though.

But I think that would be a different discussion, why someone prefers sending patches through GitHub instead of the mailing list, and where the community is the most active in getting feedback.

paper42 committed:
base-files: do not match /etc/init.d/

prerm and postinst hooks try to use services in /etc/init.d/, but the
grep also matched /etc/init.d/ which seems to be sometimes present
resulting in ugly warnings when installing or removing packages.

Signed-off-by: Michal Vasilek <michal.vasilek@nic.cz>
